### PR TITLE
Adding UPS support

### PR DESCRIPTION
**Pull request by secondsun**
See https://drive.google.com/file/d/0BzSy-Z4v4qE-aDBMbGVldGFlT28/view?usp=sharing for a demo

### CODE REVIEW

File: client/app/screens/InviteList.js [hunk 1]
```diff
@@ -83,7 +84,16 @@ export default class InviteListScreen extends React.Component {
           });
       });
 
-    // Initial load of invites
+    let reloadInvites = function(invites) {
+      this.setState({ dataSource: ds.cloneWithRows(invites) });
```
Comment (pmdarrow) [inline client/app/screens/InviteList.js]: It looks like you're reusing the code on line 78 — I'd extract it to a class method `_loadInvites` or similar and call it above and in `onDefaultMessage`.

File: client/app/screens/Loading.js [hunk 2]
```diff
@@ -59,16 +60,29 @@ export default class LoadingScreen extends React.Component {
...
+            },
             this._attemptNavigation
           );
         } else {
-          this.setState({ hasToken: false }, this._attemptNavigation);
+          this.setState(
+            {
+              hasToken: false,
+            },
+            this._attemptNavigation
```
Comment (pmdarrow) [inline client/app/screens/Loading.js]: Hmm, passing functions through state seems like an anti-pattern—what's going on here?

File: client/app/screens/InviteList.js [hunk 3]
```diff
@@ -83,7 +84,16 @@ export default class InviteListScreen extends React.Component {
...
+    }.bind(this);
+
+    DeviceEventEmitter.addListener('onDefaultMessage', function(event) {
```
Comment (pmdarrow) [inline client/app/screens/InviteList.js]: It looks like this method of sending events from Java to JS is deprecated and no longer documented — see https://github.com/facebook/react-native/pull/12867/commits/b11f8f4936459c104b4fccb3da5eef1d53aa76f7. Not sure what the alternative is though...

File: client/app/screens/Loading.js [hunk 4]
```diff
@@ -59,16 +60,29 @@ export default class LoadingScreen extends React.Component {
...
           this.setState(
-            { hasToken: true, token: tokens['access_token'] },
+            {
+              hasToken: true,
+              token: tokens['access_token'],
+            },
             this._attemptNavigation
           );
         } else {
-          this.setState({ hasToken: false }, this._attemptNavigation);
```
Comment (pmdarrow) [inline client/app/screens/Loading.js]: This file was a lot more readable with these `setState` calls on single lines IMO. Do you mind rolling this back?

File: server/api/controllers/inviteController.js [hunk 5]
```diff
@@ -2,6 +2,13 @@
 
 var mongoose = require('mongoose');
 var Invite = require('../models/inviteModel');
+var agSender = require('unifiedpush-node-sender');
+
+const settings = {
+    url: "http://ups-server-snapscreen.74.207.224.48.nip.io/ag-push/",
+    applicationId: "5bc4aebd-d991-40fe-90f8-e17ed2e6c453",
+    masterSecret: "2aa47426-29c5-41ea-970f-32efb1a0f8b8"
```
Comment (pmdarrow) [inline server/api/controllers/inviteController.js]: 😳

Comment (secondsun) [reply server/api/controllers/inviteController.js]: These are already extracted in the property_extraction branch.

File: client/app/screens/InviteList.js (hunk 3 above)
Comment (secondsun) [reply client/app/screens/InviteList.js]: So there is RCTEventEmitter in iOS which looks like the "golden scenario" but similar APIs don't exist in Android.  I dug into various other event emitter APIs but they all seem to have a similar "what docs" problem

File: client/app/screens/Loading.js (hunk 4 above)
Comment (secondsun) [reply client/app/screens/Loading.js]: Fixed, but yarn form splits it back onto two lines.

**Comment by secondsun** [comment]
@pmdarrow I think I've addressed everything one way or another.

File: client/app/screens/Loading.js (hunk 4 above)
Comment (pmdarrow) [reply client/app/screens/Loading.js]: Weird, I tried it out and it didn't reformat them for me, but maybe I didn't have the exact same layout.

**Pull request merged**